### DISCUSSION TITLE
How to use MOOSE to generate a layered square with circular void in the middle?

### DISCUSSION THREAD

**Discussion by Yulongggggg** [idaholab/moose#29045] [Q&A Meshing]
Dear developers, I am trying to use MOOSE to generate a mesh, which is a square with a circular void inside, can u tell me how to do it? Also, the square should be layered, i.e., composed of different blocks(rectangle shape), and how to assign different boundary names (apply to the boundary condition block. e.g., 'inner')

**Comment by GiudGiud** [comment]
>  I am trying to use MOOSE to generate a mesh, which is a square with a circular void inside, can u tell me how to do it?

You can use this from the Reactor module
https://mooseframework.inl.gov/source/meshgenerators/PinMeshGenerator.html
or this
https://mooseframework.inl.gov/source/meshgenerators/ConcentricCircleMeshGenerator.html
with `has_outer_square` set to true

then delete the central cirlcle using a BlockDeletionGenerator

> and how to assign different boundary names (apply to the boundary condition block. e.g., 'inner')

Once the blocks are set you can use this
https://mooseframework.inl.gov/source/meshgenerators/SideSetsBetweenSubdomainsGenerator.html

**Reply by Yulongggggg** [reply]
Thank you! Could you please tell me how to generate a layered square mesh with inclusion inside? Which looks like this
<img width="770" alt="Screenshot 2024-10-29 at 2 34 47 PM" src="https://github.com/user-attachments/assets/a52f48d7-0d0e-4af6-9524-fa25ebceb4c5">

**Reply by GiudGiud** [reply]
with the mesh generators I mentioned you should get a good start. 
You can add subdomains 1 by 1 using this generator
https://mooseframework.inl.gov/source/meshgenerators/ParsedSubdomainMeshGenerator.html

how many layered subdomains do you need?
